core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -474,13 +477,13 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
    * Spark fair scheduler pool.
    */
   def setLocalProperty(key: String, value: String) {
-    if (localProperties.get() == null) {
-      localProperties.set(new Properties())
+    if (localProperties.get().isEmpty) {
+      localProperties.set(Some(new Properties()))
```

Comment:
By the way, I think this change would significantly shrink the size of this patch, too, since none of the internal methods' type signatures would need to change.